plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java

```diff
@@ -0,0 +1,368 @@
+
+                        for (Map.Entry<String, String> entry : outputMapping.entrySet()) {
+                            String originalModelOutputFieldName = entry.getKey(); // response
+                            String newModelOutputFieldName = entry.getValue(); // inference_result
+                            appendFieldValue(modelTensorOutput, originalModelOutputFieldName, newModelOutputFieldName, ingestDocument);
+                        }
+                    }
+                    process_predictions(ingestDocument, handler, processInputMap, processOutputMap, i + 1, round);
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    handler.accept(null, e);
+                    return;
+                }
+            });
+        }
+
+    }
+
+    private void getMappedModelInputFromDocuments(
+        IngestDocument ingestDocument,
+        Map<String, String> modelParameters,
+        String originalFieldName,
+        String ModelInputFieldName
+    ) {
+        String originalFieldPath = getFieldPath(ingestDocument, originalFieldName);
+        if (originalFieldPath != null) {
+            Object originalFieldValue = ingestDocument.getFieldValue(originalFieldPath, Object.class);
+            String originalFieldValueAsString = toString(originalFieldValue);
+            updateModelParameters(ModelInputFieldName, originalFieldValueAsString, modelParameters);
+        }
+        // check for nested array
+        else {
+            if (originalFieldName.contains(".")) {
+
+                Map<String, Object> sourceObject = ingestDocument.getSourceAndMetadata();
+                Configuration suppressExceptionConfiguration = Configuration
+                    .builder()
+                    .options(Option.SUPPRESS_EXCEPTIONS, Option.DEFAULT_PATH_LEAF_TO_NULL, Option.ALWAYS_RETURN_LIST)
+                    .build();
+                String rewriteDotPathForNestedObject = findDotPathForNestedObject(sourceObject, originalFieldName);
+                ArrayList<Object> fieldValueList = JsonPath
+                    .using(suppressExceptionConfiguration)
+                    .parse(sourceObject)
+                    .read(rewriteDotPathForNestedObject);
+
+                String originalFieldValueAsString = toString(fieldValueList);
+                updateModelParameters(ModelInputFieldName, originalFieldValueAsString, modelParameters);
+            }
+        }
+    }
+
+    /**
+     * support multiple document fields to map to the same model input fields,
+     * check if the key existed, then append to existed value to be a list.
+     */
+    private void updateModelParameters(String ModelInputFieldName, String originalFieldValueAsString, Map<String, String> modelParameters) {
+
+        if (modelParameters.containsKey(ModelInputFieldName)) {
+            Object existingValue = modelParameters.get(ModelInputFieldName);
```

Comment:
why `existingValue` is an Object type? isn't `modelParameters` parameter declared as `Map<String, String>`?